Проектная работа №2. Применяем дженерики.

morechiba opened the pull request.

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

morechiba committed:
Первая версия с новыми дженериками

morechiba committed:
Лишняя ;

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,26 @@
+
+    public boolean theSameAs(T t){
+        return this.hashCode() == t.hashCode();
+    }
```
Эта функция лишняя, её можно удалить

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.hashCode() == newItem.hashCode();
```
Здесь нужно вызвать методы интерфейса `ListDiffInterface` у обоих аргументов функции:
```suggestion
        return oldItem.theSameAs(newItem);
```

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/FilterContactTypeAdapter.java:
```diff
@@ -17,16 +17,27 @@
+    @Override
+    public boolean theSameAs(FilterContactTypeUi f){
+        return this.contactType== f.getContactType();
+    }
```
Сам адаптер не должен реализовывать интерфейс `ListDiffInterface`

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +22,11 @@ public boolean isSelected() {
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi filterContactTypeUi) {
+        return false;
```
Здесь нужно сравнить текущий объект с объектом из аргумента:
```suggestion
       return this.getContactType() == filterContactTypeUi.getContactType();
```

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/main/ContactAdapter.java:
```diff
@@ -23,12 +23,20 @@
+    @Override
+    public boolean theSameAs(ContactUi c){
+        return this.hashCode() == c.hashCode();
+    };
```
Здесь тоже не нужно наследовать интерфейс `ListDiffInterface`

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/main/ContactUi.java:
```diff
@@ -41,6 +42,11 @@ public List<ContactType> getTypes() {
 
+    @Override
+    public boolean theSameAs(ContactUi contactUi) {
+        return false;
```
Здесь нужно сравнить текущий объект с объектом из аргумента:
```suggestion
       return this.hashCode() == contactUi.hashCode();
```

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeAdapter.java:
```diff
@@ -17,13 +17,21 @@
+    @Override
+    public boolean theSameAs(SortTypeUI s){
+        return this.sortType == s.getSortType();
+    };
```
Адаптер не должен реализовывать интерфейс `ListDiffInterface`

ArturNurtdinov commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java:
```diff
@@ -22,6 +23,11 @@ public boolean isSelected() {
 
+    @Override
+    public boolean theSameAs(SortTypeUI sortTypeUI) {
+        return false;
```
```suggestion
        return this.getSortType() == sortTypeUI.getSortType();
```

morechiba committed:
убрана реализация интерфейса ListDiffInterface в адаптерах, убран лишний метод theSameAs()
переделан метод theSameAs() в классах, описывающих элемент списка
в классе BaseListDiffCallback убран лишний метод

morechiba commented:
Исправила ошибки в файлах и только потом увидела, что можно было подтверждать изменения :)

ArturNurtdinov commented:
> Исправила ошибки в файлах и только потом увидела, что можно было подтверждать изменения :)

Привет, это не обязательно, но можно, да)
